pkg/build/builder/source.go

```diff
@@ -77,7 +77,9 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 func checkRemoteGit(url string, timeout time.Duration) error {
 	glog.V(4).Infof("git ls-remote --heads %q", url)
 	cmd := exec.Command("git", "ls-remote", "--heads", url)
-	cmd.Env = []string{"GIT_ASKPASS=/bin/true", "GIT_SSH=" + os.Getenv("GIT_SSH")}
+	env := os.Environ()
+	env = append(env, "GIT_ASKPASS=/bin/true")
```

Comment:
Interesting… is the intent to fail fast here if an auth prompt is needed to prevent the clone from hanging? If we're not setting GIT_ASKPASS during the actual clone, what is it buying us here?